src/plain.jl

```diff
@@ -269,17 +269,17 @@ type UTF8Char<:CharType
     c::UInt8
 end
 length(c::UTF8Char) = 1
-chartype(::Type{Compat.ASCIIString}) = ASCIIChar
```

Comment:
@StefanKarpinski, if you think this works you might want to try running the JLD and JLDArchives tests.

The problem @yuyichao is noting is that these are still separate types on julia-0.3 and julia-0.4. Changing it to `Type{String}` seems like it will only match the non-leaftype `ByteString` and throw a `MethodError` for a concretely-type `ASCIIString` or `UTF8String`.